c-sharp.md

```diff
@@ -5,70 +5,37 @@ tags = ["c#", "csharp", "dotnet", ".net", "webassembly"]
 
 ## Available Implementations
 
-The .NET ecosystem has long had support for browser-side WebAssembly via the [Blazor toolkit](https://dotnet.microsoft.com/en-us/apps/aspnet/web-apps/blazor).
+The .NET ecosystem has long had support for browser-side WebAssembly via the [Blazor toolkit](https://dotnet.microsoft.com/en-us/apps/aspnet/web-apps/blazor). As of .NET 7, it has support for WebAssembly as a compile target.
 
-The team at Microsoft is rapidly building out a number of WebAssembly tools. In 2022, it seems reasonable to expect that Microsoft will have new tooling for WebAssembly. C# compiled using the .NET toolchain will have access to these WebAssembly features.
-
-At the time of this writing, the closest we have seen to a full WASI runtime is Steve Sanderson's example [.NET WASI Runtime](https://github.com/SteveSandersonMS/dotnet-wasi-runtime).
+The team at Microsoft is rapidly building out a number of WebAssembly tools. .NET 7 includes support for compiling to WebAssembly. Read [Ivan Towlson's excellent tutorial](https://www.fermyon.com/blog/dotnet-wasi) on building a WebAssembly app with .NET.
 
 In addition, browser-based support also seems to be available in the [Elements compiler](https://www.elementscompiler.com/elements/)
 
-## Pros and Cons
-
-Things we like:
-
-- Blazor is enjoying a surge of popularity in the browser
-- The .NET WASI Runtime looks excellent
-
-For the most part, though, it is too early to speculate on the rest of the .NET tooling.
-
 ## Example
 
-Our example uses the [.NET WASI Runtime](https://github.com/SteveSandersonMS/dotnet-wasi-runtime). This is a preview and requires the supporting tools and build steps described in that repository.
-
->> All of our examples follow [a documented pattern using common tools](/wasm-languages/about-examples).
+You will need to be running .NET 7 and also have the [WASI SDK](https://github.com/SteveSandersonMS/dotnet-wasi-sdk).
 
-The example requires a project file `ConsoleApp.csproj` that imports the .NET WASI SDK:
+>> This example is derived from [Ivan's blog post](https://www.fermyon.com/blog/dotnet-wasi). Other examples of C# and F# can be found [here](https://github.com/fermyon/spin-kitchensink/)
 
-```xml
-<Project Sdk="Microsoft.NET.Sdk">
-
-    <!--
-        As this is a preview, you will need to specify the path at which you
-        checkout out the dotnet-wasi-runtime repo.
-    -->
-	<Import Project="path\to\Wasi.Sdk\build\Wasi.Sdk.props" />
-
-	<PropertyGroup>
-		<OutputType>Exe</OutputType>
-		<TargetFramework>net7.0</TargetFramework>
-		<ImplicitUsings>enable</ImplicitUsings>
-		<Nullable>enable</Nullable>
-	</PropertyGroup>
-
-	<ItemGroup>
-		<ProjectReference Include="path\to\src\Wasi.Sdk\Wasi.Sdk.csproj" ReferenceOutputAssembly="false" />
-	</ItemGroup>
-
-	<Import Project="path\to\Wasi.Sdk\build\Wasi.Sdk.targets" />
-
-</Project>
-```
+Create a new project with `dotnet new console`. Then add the WASI SDK like this: `dotnet add package Wasi.Sdk --prerelease`
```

Comment:
I think this is now `dotnet workload install wasi-experimental` (and then you run `dotnet new wasiconsole`) but I have kind of lost track of it all.  @jpflueger is likely more up-to-date than me at this point!

Reply:
That is spot on. [This repo from Steve is a good reference](https://github.com/SteveSandersonMS/spiderlightning-dotnet) for how it dotnet 8 will work, though our approach is slightly different unless we wanted to follow a similar approach to how go adds handlers during the init function.